src/borg/locking.py

```diff
@@ -21,6 +23,29 @@ def get_id():
     return _hostname, _pid, thread_id
 
 
+def check_lock_data(host, pid, thread):
+    """Check if the host, pid, thread combination corresponds to a dead process on our local node or not."""
+    if host != _hostname:
+        return False
+
+    try:
+        # This may not work in Windows.
```

Comment:
Should work; on Windows you don't really send signals, but open a process and then call various functions on the process handle. Python shouldn't be different; opening the process for a dead PID fails and should raise OSError. The error number will be different from ESRCH ...